frontend/ui/src/reducers/data/conversations/index.ts

```diff
@@ -216,7 +216,10 @@ function allReducer(
           [action.payload.identifier]: {
             id: action.payload.identifier,
             ...state.items[action.payload.identifier],
-            metadata: merge({}, state.items[action.payload.identifier]?.metadata, action.payload.metadata),
+            metadata: {
+              ...merge({}, state.items[action.payload.identifier]?.metadata, action.payload.metadata),
+              state: state.items[action.payload.identifier].metadata.state,
```

Comment:
I'm not sure I understand this: When a metadata update comes in we ignore it and keep the present value even when it's `undefined`?

Reply:
the conversation state is currently being changed twice in redux each time (first via the conversation state action and then via the websocket's metadata update). There is a delay between both updates, which creates a buggy behaviour. Maybe i'm missing something but I don't think this value needs to be updated twice by two different actions.

my idea is to ignore the websocket's metadata update for conversation state because it comes after the first update

Reply:
but then we might be ignoring some other relevant metadata updates right? like the tags from example. Do tags work with this changes?

Reply:
I'm only ignoring the metadata update for state (the changes are made for the others metadata updates like tags or display name ect.)

`metadata: {
              ...merge({}, state.items[action.payload.identifier]?.metadata, action.payload.metadata),
              state: state.items[action.payload.identifier].metadata.state,`

Reply:
@AudreyKj yes the bug is likely due to the delay. As far as I understand the first update is optimistic, but just like with the contact display name we rely on the final state to arrive via the WebSocket. 

Another thing to consider with this solution is that other users of the same inbox won't be able to see that a conversation's state has changed. 

>**To Reproduce**
Steps to reproduce the behavior:
> 1. In the Inbox UI of any Airy instance with a lot of conversations
> 2. when you go manually through them and change the status of multiple conversations (e.g. from OPEN to CLOSED) 
> 3. you will see inconsistencies happening from time to time, i.e. button not clickable or button clickable, but status change is not reflected properly in the conversation list/conversation view or when reloading the view for the relevant conversation

Were you able to reproduce the behavior? It could also be that the API calls to update the state are failing for this user somehow.

Reply:
@chrismatix yeah I was able to reproduce locally and see that inconsistencies are happening from time to time (it happens occasionally and when the user quickly toggles the state because of the delay). The API calls are working locally but it could also be that the API calls are failing for a particular user.

yeah this solution is not ideal because it will prevent state conversation change in real time - I'm researching others alternatives